### DISCUSSION TITLE
Customise theme to remove border radius on all components?

### DISCUSSION THREAD

**Discussion by j-lee8** [chakra-ui/chakra-ui#4173] [Q&A]
Is there a way to remove the border radius on all components in the theme? Also, extendTheme doesn't appear to auto-complete the different props so it's difficult to know what can be customised. 

I've tried this but it only works for Button?

```
const customTheme = extendTheme({ 
	colors: {
		brand: {

		}
	},
	components: {
		Button: {
			baseStyle: {
				rounded: 'none'
			}
		},
		Input: {
			baseStyle: {
				rounded: 'none'
			}
		}
	}
});
```

Thanks all.

**Comment by anubra266** [comment]
Hi! I think you're looking for the [withDefaultProps](https://chakra-ui.com/docs/theming/customize-theme#theme-extension-withdefaultprops) extension.

**Reply by dannyvaughton** [reply]
I don't think this works for border radius, only for variant, size & colorScheme.

**Comment by jacklee814** [comment]
Hi @JavaJamie 
I have the same request too, I checked the Input source code then found the `borderRadius` in [size](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/input.ts#L21) property, and size was under the [sizes.field](https://github.com/chakra-ui/chakra-ui/blob/3adda59f95986abfc7b4d97a855900270cde2d3c/packages/theme/src/components/input.ts#L51) property. My solution is add `field` property outer of `borderRadius`, it's working.

```js
const theme = extendTheme({
  components: {
    Input: {
      sizes: {
        lg: {
          field: {
            borderRadius: 'none',
          },
        },
      },
    },
  },
});
```

**Reply by nitedani** [reply]
I had to specify md instead of lg for it to work. I think you need to specify all sizes just to be sure it will stay working.
